LookAndFeels/foleys_Skeuomorphic.cpp

```diff
@@ -0,0 +1,240 @@
+ */
+
+namespace foleys
+{
+using namespace juce;
+
+static inline void clipToEllipse (Graphics& g, Rectangle<float>& bounds)
+{
+    Path p;
+    p.addEllipse (bounds);
+    g.reduceClipRegion (p);
+}
+
+static inline ColourGradient makeGradient (Point<float> point1,
+                                           Point<float> point2,
+                                           bool isRadial,
+                                           std::initializer_list<std::pair<double, Colour>> list)
+{
+    ColourGradient gr {};
+    gr.isRadial = isRadial;
+    gr.point1 = point1;
+    gr.point2 = point2;
+
+    for (auto pair : list)
+        gr.addColour (pair.first, pair.second);
+
+    return gr;
+}
+
+static inline void fillEllipse (Graphics& g, const Rectangle<float>& bounds, Colour colour)
+{
+    g.setColour (colour);
+    g.fillEllipse (bounds);
+}
+
+static inline void fillEllipse (Graphics& g, const Rectangle<float>& bounds, const ColourGradient& gradient)
+{
+    g.setGradientFill (gradient);
+    g.fillEllipse (bounds);
+}
+
+const Skeuomorphic::knobImages& Skeuomorphic::getKnobImages (int diameter)
+{
+    auto search = knobsBgs.find (diameter);
+    if (search != knobsBgs.end()) return search->second;
+
+    float radius = diameter * 0.5f;
+    bool isTiny  = diameter < 20;
+    bool isSmall = diameter < 60;
+
+    Point<float>     centre { radius, radius };
+    Rectangle<float> bounds { 0, 0, float (diameter), float (diameter) };
+
+    if (knobsBgs.size() > size_t (knobsMaxSize))
+        knobsBgs.clear();
+
+    Image bg = Image (Image::ARGB, diameter, diameter + 3, true);
+
+    Graphics g (bg);
+    fillEllipse (g, bounds.translated (0.0f, 1.0f), whiteA010);
+    fillEllipse (g, bounds.translated (0.0f, 2.0f), whiteA010);
+    fillEllipse (g, bounds,                         { 50, 51, 61});
+
+    auto backgroundGr = isTiny? makeGradient (centre, { centre.x, centre.y + radius }, true,
+                                          {  { 0.0, Colours::transparentBlack },
+                                             { 0.9, Colours::transparentBlack },
+                                             { 1.0, blackA122 }  }) :
+                                makeGradient (centre, { centre.x, centre.y + radius }, true,
+                                          {  { 0.0,                          Colours::transparentBlack },
+                                             { 0.52,                         blackA092 },
+                                             { 0.9,                          Colours::transparentBlack },
+                                             { double (1 - (4.0f / radius)), Colours::transparentBlack },
+                                             { 1.0,                          blackA092 } });
+    fillEllipse (g, bounds, backgroundGr);
+
+
+
+    bounds.reduce (diameter * 0.15f, diameter * 0.15f);
+    float fgRadius = bounds.getWidth() * 0.5f;
+
+    Image fg = Image (Image::ARGB, diameter, diameter, true);
```

Comment:
Is the same as
`auto fg = Image (Image::ARGB, diameter, diameter, true);`
or
`Image fg (Image::ARGB, diameter, diameter, true);`

Reply:
Makes sense too